contract-deployment/utils/deployment_utils.go

```diff
@@ -0,0 +1,146 @@
+package utils
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"log"
+	"math/big"
+	"os"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/pkg/errors"
+)
+
+const (
+	// Roughly 3,010,000 gas needed to deploy contract. Padded to account for possible additions
+	contractCreationGasLimit = uint64(4000000)
+
+	// R and S values to use in a keyless transaction signature.
+	// The values do not technically need to be the same when using Nick's method, but the AvalancheGo
+	// APIs by default only allow legacy transactions to be broadcast if they have the same R and S values,
+	// which is used as a heuristic to identify (and allow) Nick's method transactions.
+	rsValueHex                       = "3333333333333333333333333333333333333333333333333333333333333333"
+	contractCreationTxFileName       = "UniversalTeleporterDeployerTransaction.txt"
+	contractCreationAddrFileName     = "UniversalTeleporterDeployerAddress.txt"
+	universalContractAddressFileName = "UniversalTeleporterMessengerContractAddress.txt"
+)
+
+var (
+	vValue                   = big.NewInt(27)            // Must be less than 35 to be considered non-EIP155
+	contractCreationGasPrice = big.NewInt(2500000000000) // 2500 nAVAX/gas
+)
+
+type byteCodeObj struct {
+	Object string `json:"object"`
+}
+
+type byteCodeFile struct {
+	ByteCode byteCodeObj `json:"bytecode"`
+}
+
+func DeriveEVMContractAddress(sender common.Address, nonce uint64) (common.Address, error) {
+	type AddressNonce struct {
+		Address common.Address
+		Nonce   uint64
+	}
+	addressNonce := AddressNonce{sender, nonce}
+	rlpEncoded, err := rlp.EncodeToBytes(addressNonce)
+	if err != nil {
+		return common.Address{}, errors.Wrap(err, "Failed to RLP encode address and nonce value.")
+	}
+	hash := crypto.Keccak256Hash(rlpEncoded)
+	return common.HexToAddress(fmt.Sprintf("0x%x", hash.Bytes()[12:])), nil
+}
+
+// Constructs a keyless transaction using Nick's method
+// Optionally writes the transaction, deployer address, and contract address to file
+// Returns the transaction bytes, deployer address, and contract address
+func ConstructKeylessTransaction(byteCodeFileName string, writeFile bool) ([]byte, common.Address, common.Address, error) {
```

Comment:
It looks like `writeFile` is not properly used/checked in the implementation.

Reply:
Whoops, fixed now.